server/src/main/java/org/opensearch/cluster/routing/allocation/allocator/LocalShardsBalancer.java

```diff
@@ -799,8 +828,23 @@ void allocateUnassigned() {
         int secondaryLength = 0;
         int primaryLength = primary.length;
         ArrayUtil.timSort(primary, comparator);
+        if (logger.isTraceEnabled()) {
+            logger.trace("Staring allocation of [{}] unassigned shards", primaryLength);
+        }
         do {
             for (int i = 0; i < primaryLength; i++) {
+                if (timedOutFunc.apply(System.nanoTime())) {
+                    // TODO - maybe check if we can allow wait for active shards thingy bypass this condition
```

Comment:
Thanks for catching this. Lets open a tracking issue?

Reply:
Yes. Created here - https://github.com/opensearch-project/OpenSearch/issues/15433